### PR TITLE
Separate VSCode Windows and macOS tutorials

### LINKED ISSUE #137: Separate VSCode Windows and macOS tutorials
It's very easy to make mistakes on the VSCode tutorial by accidentally following a step meant for the wrong OS.  While I don't usually like to copy / paste things, I think that would really reduce student frustration.

### LINKED ISSUE #148: VS Code Install Extensions Command Ordering
The commands to install VS Code extensions are in the wrong order (and I believe only recently is the ordering significant.)

Quick Version:

I believe everything is fixed if we change the ordering to:
```console
code --install-extension ms-vscode-remote.remote-wsl
code --install-extension ms-vscode.cpptools
```

This installs WSL extension first, which ensures the subsequent C++ extension is installed on the WSL side as it should be.

Long Version:

For some context, once the WSL extension is installed, additional extensions like the C++ extension may be installed locally in Windows VS Code or remotely in WSL VS Code.

I *think* the problem is that running `code` from a bash terminal no longer installs the WSL extension automatically if it wasn't already installed. Instead, it will just affect the Windows VS Code, including installing extensions to there.

We give these commands to install and verify extensions:

image

It's unclear whether they should be run in PowerShell or in bash.

If they are run in Powershell, the effect is:
1. Install C++ extension in Windows VS Code but not WSL VS Code
2. Install WSL extension in Windows VS Code
3. Listed extensions match what's show in the tutorial, but this is misleading since C++ extension is not installed in WSL VS Code.

If they are run in bash, the effect is:
1. Install C++ extension in Windows VS Code
2. Install WSL extension in Windows VS Code. Subsequent `code` commands begin to refer to WSL VS Code.
3. Listed extensions do not include either shown in the tutorial, since no extensions have been installed in WSL VS Code.
4. Student may try installing both again. Now C++ gets installed to WSL VS Code. A weird "dummy" remote WSL extension gets installed into WSL VS Code.
5. Listed extensions match the tutorial.

### PR DESCRIPTION
**Pull request by awdeorio**
Copy VS Code tutorial into separate Windows and macOS versions.  Hopefully this will make the VS Code tutorial easier to follow.

Closes #137, Closes #148

### CODE REVIEW

**Comment by awdeorio** [comment]
This is ready for review after a resolving a very tricky merge conflict from #147

File: docs/setup_vscode_macos.md
Comment (amirkamil) [inline docs/setup_vscode_macos.md:72]: Perhaps remove "AKA File Explorer", since that's Windows.

File: docs/setup_vscode_wsl.md
Comment (amirkamil) [inline docs/setup_vscode_wsl.md:78]: Perhaps remove "Finder AKA", since that's macOS.

File: docs/setup_vscode_wsl.md
Comment (amirkamil) [inline docs/setup_vscode_wsl.md:540]: I think this entire section can be deleted, since we aren't using lldb on Windows.

File: docs/setup_vscode_macos.md
Comment (amirkamil) [inline docs/setup_vscode_macos.md:432]: I don't think this is relevant to macOS, since it uses lldb rather than gdb.

File: docs/setup_vscode_wsl.md
Comment (jamesjuett) [inline docs/setup_vscode_wsl.md:56]: This feels redundant since it immediately follows the previous section where we have students do the same thing. I suggest consolidating them. This would also help make it more clear that the Ubuntu/bash terminal (only mentioned in the previous section) is what they need to use for installing extensions (i.e. not PowerShell).

File: docs/setup_vscode_wsl.md
Comment (jamesjuett) [inline docs/setup_vscode_wsl.md:64]: Reverse the ordering here to fix #148. Install `ms-vscode-remote.remote-wsl` first, then `ms-vscode-cpptools`.

File: docs/setup_vscode_wsl.md
Comment (jamesjuett) [inline docs/setup_vscode_wsl.md:114]: This will run into some issues since it opens that folder in "Windows" VS Code.

Instead `View` > `Command Palette` (or ctrl+shift+p). Then search for "WSL: Open Folder in WSL".

File: docs/setup_vscode_wsl.md
Comment (jamesjuett) [inline docs/setup_vscode_wsl.md:128]: I think this "open your project folder" part can be removed since it was just covered immediately above. (If it is kept, it needs to be fixed in the same way as the above to use "open folder in WSL" instead.)

**Review by jamesjuett — CHANGES_REQUESTED** [review]
I left a few specific comments.

In general, many of the screenshots are still "Mac" in superficial ways, but I still have the screenshots I overzealously tried to replace in the other PR - so if it's alright I can go ahead and push them here as well.

File: docs/setup_vscode_macos.md
Comment (awdeorio) [reply docs/setup_vscode_macos.md]: Fixed 9d20b6ba28920b124871251ec8e955bf40f0b43b

File: docs/setup_vscode_wsl.md
Comment (awdeorio) [reply docs/setup_vscode_wsl.md]: Fixed 9d20b6ba28920b124871251ec8e955bf40f0b43b

File: docs/setup_vscode_macos.md
Comment (awdeorio) [reply docs/setup_vscode_macos.md]: Fixed 9d20b6ba28920b124871251ec8e955bf40f0b43b

File: docs/setup_vscode_wsl.md
Comment (awdeorio) [reply docs/setup_vscode_wsl.md:128]: The reason this is here is for students visiting the section via a fragment link from a project spec, e.g.,  https://eecs280staff.github.io/tutorials/setup_vscode.html#add-new-files

Comment (awdeorio) [reply docs/setup_vscode_wsl.md:128]: Uh oh.  I just realized that this PR will break all the fragment links into the VSCode tutorial!

File: docs/setup_vscode_wsl.md
Comment (awdeorio) [reply docs/setup_vscode_wsl.md]: Fixed 28f2d0ae66ebac0baada4b42b27f360cd847bc48

File: docs/setup_vscode_wsl.md
Comment (awdeorio) [reply docs/setup_vscode_wsl.md]: Fixed 28f2d0ae66ebac0baada4b42b27f360cd847bc48

File: docs/setup_vscode_wsl.md
Comment (awdeorio) [reply docs/setup_vscode_wsl.md]: Fixed 28f2d0ae66ebac0baada4b42b27f360cd847bc48

File: docs/setup_vscode_wsl.md
Comment (awdeorio) [reply docs/setup_vscode_wsl.md:122]: Update: I create an issue in each project repo to address this problem.

Comment (jamesjuett) [reply docs/setup_vscode_wsl.md:122]: I don't have a strong preference, but it will need to be adjusted to the "open folder in WSL" instructions as with the previous instance.

**Review by jamesjuett — CHANGES_REQUESTED** [review]

**Review by amirkamil — APPROVED** [review]
I don't want to hold this up to handle orthogonal changes. I think this LGTM as is.

File: docs/setup_vscode_wsl.md
Comment (jamesjuett) [reply docs/setup_vscode_wsl.md:66]: Amir and I chatted separately. I'll make an issue for the fix to the `--list-extensions` part so that can be handled separately as well.

File: docs/setup_vscode_wsl.md
Comment (jamesjuett) [reply docs/setup_vscode_wsl.md:122]: I'll make a separate issue for this fix so that we can get the PR through.

**Review by jamesjuett — APPROVED** [review]

**Pull request merged**
